src/libsyncengine/jobs/network/upload_session/driveuploadsession.cpp

```diff
@@ -0,0 +1,113 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "driveuploadsession.h"
+#include "utility/utility.h"
+
+namespace KDC {
+
+DriveUploadSession::DriveUploadSession(int driveDbId, std::shared_ptr<SyncDb> syncDb, const SyncPath &filepath,
+                                       const SyncName &filename, const NodeId &remoteParentDirId, SyncTime modtime,
+                                       bool liteSyncActivated, uint64_t nbParalleleThread /*= 1*/)
+    : AbstractUploadSession(filepath, filename, nbParalleleThread),
+      _driveDbId(driveDbId),
+      _syncDb(syncDb),
+      _modtimeIn(modtime),
+      _liteSyncActivated(liteSyncActivated),
+      _remoteParentDirId(remoteParentDirId) {
+    _uploadSessionType = UploadSessionType::Standard;
+}
+
+DriveUploadSession::~DriveUploadSession() {
+    if (_vfsForceStatus && !_vfsForceStatus(getFilePath(), false, 100, true)) {
+        LOGW_WARN(getLogger(), L"Error in vfsForceStatus: " << Utility::formatSyncPath(getFilePath()).c_str());
+    }
+}
+
+bool DriveUploadSession::runJobInit() {
+    return true;
+}
+
+std::shared_ptr<UploadSessionStartJob> DriveUploadSession::createStartJob() {
+    return std::make_shared<UploadSessionStartJob>(UploadSessionType::Standard, _driveDbId, getFileName(), getFileSize(),
+                                                   _remoteParentDirId, getTotalChunks());
+}
+
+std::shared_ptr<UploadSessionChunkJob> DriveUploadSession::createChunkJob(const std::string &chunckContent, uint64_t chunkNb,
+                                                                          std::streamsize actualChunkSize) {
+    return std::make_shared<UploadSessionChunkJob>(UploadSessionType::Standard, _driveDbId, getFilePath(), getSessionToken(),
+                                                   chunckContent, chunkNb, actualChunkSize, jobId());
+}
+
+std::shared_ptr<UploadSessionFinishJob> DriveUploadSession::createFinishJob() {
+    return std::make_shared<UploadSessionFinishJob>(UploadSessionType::Standard, _driveDbId, getFilePath(), getSessionToken(),
+                                                    getTotalChunkHash(), getTotalChunks(), _modtimeIn);
+}
+
+std::shared_ptr<UploadSessionCancelJob> DriveUploadSession::createCancelJob() {
+    return std::make_shared<UploadSessionCancelJob>(UploadSessionType::Standard, _driveDbId, getFilePath(), getSessionToken());
+}
+
+// Set VFS callbacks
+bool DriveUploadSession::prepareChunkJob(const std::shared_ptr<UploadSessionChunkJob> &chunkJob) {
+    if (_liteSyncActivated) {
+        chunkJob->setVfsUpdateFetchStatusCallback(_vfsUpdateFetchStatus);
+        chunkJob->setVfsSetPinStateCallback(_vfsSetPinState);
+        chunkJob->setVfsForceStatusCallback(_vfsForceStatus);
```

Comment:
Those callbacks are not used in ChunkJob

Reply:
[5ea4baf](https://github.com/Infomaniak/desktop-kDrive/pull/139/commits/5ea4bafd77626c3ccb6d528cc37133e50087ddb1)